File: cpp/oop/diamondInheritance/Animal.cpp
#include <iostream>
#include <string>

#include "Animal.h"

using namespace std;

Animal::Animal(string name) {
    cout << "Animal::Animal: name=" << name << endl;
    mName = name;
}

Animal::~Animal() {
    cout << "Animal::~Animal" << endl;
}

string Animal::getName() {
    cout << "Animal::getName: mName=" << mName << endl;
    return mName;
}

void Animal::setName(string name) {
    cout << "Animal::setName: name=" << name << endl;
    mName = name;
}

File: java/oop/diamondInheritance/Animal.java
public class Animal {
    private String mName;

    public Animal(String name) {
	System.out.println("[D] Animal::Animal: name=[" + name + "]");
	mName = name;
    }

    public String getName() {
	System.out.println("[D] Animal::getName: mName=[" + mName + "]");
	return mName;
    }

    public void setName(String name) {
	System.out.println("[D] Animal::setName: name=[" + name + "]");
	mName = name;
    }
}

